src/app/components/link-editor-popup/index.tsx

```diff
@@ -0,0 +1,154 @@
+  x: number;
+}
+
+export const LinkEditorPopup: React.FC<LinkEditorPopupProps> = ({ editorView, node, onClose, x, y }) => {
+  const classes = useLinkEditorStyles();
+  const [link, setLink] = useState<string>(node ? node.attrs.href : '');
+
+  const handleOnChange = useCallback(
+    (event: SyntheticEvent) => {
+      setLink(event.target['value']);
+    },
+    [setLink]
+  );
+
+  const handleLaunchClick = useCallback(() => {
+    window.open(link, '_blank');
+  }, [link]);
+
+  const handleClearField = useCallback(() => setLink(''), [setLink]);
+
+  const handleApplyClick = useCallback(() => {
+    editorView.focus();
+    onClose(link);
+  }, [editorView, link, onClose]);
+
+  const handleOnClose = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  if (!editorView) {
+    return <></>;
+  }
+
+  return (
+    <Popover
+      open={true}
+      onClose={handleOnClose}
+      anchorReference="anchorPosition"
+      anchorPosition={{ top: y, left: x }}
+      anchorOrigin={{
+        vertical: 'top',
+        horizontal: 'left'
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'left'
+      }}
+    >
+      <div className={classes.container}>
+        <TextField
+          fullWidth
+          classes={{ root: classes.textField }}
+          name="link"
+          label="Link"
+          InputLabelProps={{ shrink: true }}
+          variant="outlined"
+          value={link}
+          onChange={handleOnChange}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <span className={classes.clearTextButton} onClick={handleClearField}>
+                  <CloseIcon />
+                </span>
+              </InputAdornment>
+            )
+          }}
+        />
+        <IconButton size="small" onClick={handleLaunchClick} classes={{ root: classes.launchButton }}>
+          <LaunchIcon fontSize="small" />
+        </IconButton>
+        <ActionButton variant="primaryContained" onClick={handleApplyClick} title="Apply" />
+      </div>
+    </Popover>
+  );
+};
+
+export class LinkNodeView implements NodeView {
+  dom?: HTMLAnchorElement;
+  linkEditorContainer: HTMLDivElement;
+  linkEditorView: HTMLElement;
+
+  constructor(private node: ProsemirrorNode, private view: EditorView) {
+    this.dom = document.createElement('a');
+    this.dom.style.cursor = 'pointer';
+    this.dom.setAttribute('href', this.node.attrs.href);
+    this.dom.addEventListener('click', this.open.bind(this));
+  }
+
+  open(event?: MouseEvent) {
+    event && event.preventDefault();
+
+    const { from } = this.view.state.selection;
+    const start = this.view.coordsAtPos(from) as DOMRect;
+
+    this.linkEditorContainer = this.view.dom.parentNode.appendChild(document.createElement('div'));
+    this.linkEditorContainer.style.position = 'absolute';
+    this.linkEditorContainer.style.zIndex = '10';
+
+    const x = start.left + LINK_EDITOR_MAX_WIDTH / 2;
```

Comment:
There is something wrong here, as if I click on the link multiple times the LinkEditorPopup jumps all over the place when it should be anchored to this node.

Reply:
Is it happening in Chrome?

Reply:
Yeah, not tried another browser though.